ddtrace/propagation/http.py

```diff
@@ -241,10 +243,21 @@ def _inject(span_context, headers):
             headers[HTTP_HEADER_TRACE_ID] = str(span_context.trace_id)
 
         headers[HTTP_HEADER_PARENT_ID] = str(span_context.span_id)
+
         sampling_priority = span_context.sampling_priority
+
+        if (
+            asm_config._appsec_standalone_enabled
+            and APPSEC.PROPAGATION_HEADER not in span_context._meta
+            and span_context._previous_sampling_priority
+            and sampling_priority is not None
+            and sampling_priority < span_context._previous_sampling_priority
+        ):
+            headers[HTTP_HEADER_SAMPLING_PRIORITY] = str(span_context._previous_sampling_priority)
```

Comment:
Can we add a comment to explain what/why we're doing this? I'm a bit confused by the switching out of sampling_priority values.

Reply:
Yeah, sorry it's better explained in the RFC, but long story short, the goal would be to drop the current local span while keep propagating the original sampling priority.

Since setting the MANUAL_DROP tag mainly changes the sampling priority, we need to store the previous value somewhere and use it just when injecting the new headers.